test/e2e/specs/editor/plugins/block-variations.spec.js

```diff
@@ -0,0 +1,220 @@
+		await requestUtils.activatePlugin( 'gutenberg-test-block-variations' );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin(
+			'gutenberg-test-block-variations'
+		);
+	} );
+
+	test( 'Search for the overridden default Quote block', async ( {
+		page,
+	} ) => {
+		await page
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
+
+		await page
+			.getByRole( 'region', { name: 'Block Library' } )
+			.getByRole( 'searchbox', {
+				name: 'Search for blocks and patterns',
+			} )
+			.fill( 'Quote' );
+
+		await expect(
+			page
+				.getByRole( 'listbox', { name: 'Blocks' } )
+				.getByRole( 'option', { name: 'Quote', exact: true } )
+		).toBeHidden();
+		await expect(
+			page
+				.getByRole( 'listbox', { name: 'Blocks' } )
+				.getByRole( 'option', { name: 'Large Quote' } )
+		).toBeVisible();
+	} );
+
+	test( 'Insert the overridden default Quote block variation', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '/Large Quote' );
+		await page.keyboard.press( 'Enter' );
+
+		await expect(
+			editor.canvas.getByRole( 'document', { name: 'Block: Quote' } )
+		).toHaveClass( /is-style-large/ );
+	} );
+
+	test( 'Search for the Paragraph block with 2 additional variations', async ( {
+		page,
+	} ) => {
+		await page
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
+
+		await page
+			.getByRole( 'region', { name: 'Block Library' } )
+			.getByRole( 'searchbox', {
+				name: 'Search for blocks and patterns',
+			} )
+			.fill( 'Paragraph' );
+
+		await expect(
+			page
+				.getByRole( 'listbox', { name: 'Blocks' } )
+				.getByRole( 'option' )
+		).toHaveText( [ 'Paragraph', 'Success Message', 'Warning Message' ] );
+	} );
+
+	test( 'Insert the Success Message block variation', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '/Heading' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '/Success Message' );
+		await page.keyboard.press( 'Enter' );
+
+		await expect(
+			editor.canvas.getByRole( 'document', { name: 'Paragraph block' } )
+		).toHaveText( 'This is a success message!' );
+	} );
+
+	test( 'Pick the additional variation in the inserted Columns block', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '/Columns' );
+		await page.keyboard.press( 'Enter' );
+
+		await editor.canvas
+			.getByRole( 'list', { name: 'Block variations' } )
+			.getByRole( 'button', { name: 'Four columns' } )
+			.click();
+
+		await expect(
+			editor.canvas
+				.getByRole( 'document', { name: 'Block: Columns' } )
+				.getByRole( 'document' )
```

Comment:
Should we add `{ name: /Block: Column \([1-4] of 4\)/ }` here?

Reply:
I think it's okay to have this broad locator. We just want to check the number of inner blocks.